### DISCUSSION TITLE
Apple arm64 version

### DISCUSSION THREAD

**Discussion by shaycormac** [pawelsalawa/sqlitestudio#4206] [Q&A] [answered]
Hello, first of all, thanks for open source such an excellent software, I would like to ask, there is a plan to adapt the apple M1?

**Comment by pawelsalawa** [comment, marked as answer]
Yes. As soon as GitHub Actions support M1: https://github.com/actions/virtual-environments/issues/2187

**Reply by shaycormac** [reply]
glad to hear that! thanks!

**Comment by ACTCD** [comment]
@pawelsalawa 

I found [this project](https://github.com/kramred/ungoogled-chromium-macos/actions) already using GitHub Actions a year ago to build the `AArch64/ARM64 (Apple Silicon)` and `x86-64` versions of the `Chromium project` respectively.

Is it possible for you to build `Universal` or `ARM64` versions `SQLiteStudio` through GitHub Actions with reference to how it works?

Thank you so much for such a great open source software, I really hope it will also run natively on Apple Silicon devices.

**Reply by pawelsalawa** [reply]
They don't use any GitHub feature that builds the arm64. It'a handled somehow by their compilation/building tools, which I'm not familiar with and I won't be able to do the same here. Most probably the arm64 build of sqlitestudio will be released, when github actions support it.

**Reply by ACTCD** [reply]
Okay, I see that GitHub Actions seems to have officially put on the roadmap for the last month, maybe this year they will officially launch. Thanks for your evaluation and reply.

**Comment by irmo-de** [comment]
Pre-build AArch64/ARM64 (Apple Silicon) binaries

I have built an Apple Silicon version. I noticed a 30-40% increase in speed with larger databases ❤. 

Download the prebuilt .DMG file here:
https://github.com/irmo-de/sqlitestudio-osx-arm64

**Reply by shaycormac** [reply]
cool ！！

**Reply by DarthMDev** [reply]
App crashes on startup can you help

**Reply by DarthMDev** [reply]
https://pastebin.com/0V7sLWtd

**Reply by irmo-de** [reply]
> https://pastebin.com/0V7sLWtd

You have to codesign it with:
codesign -sign SQLiteStudio.app

**Comment by scorpio810** [comment]
@irmo-de

 ```

./compile_build_bundle.sh




/moc_cfgcategory.o ./moc_cfgentry.o ./moc_signalwait.o ./moc_blockingsocket.o ./moc_threadwitheventloop.o ./moc_blockingsocketprivate.o ./moc_sqliteextensionmanager.o ./moc_lazytrigger.o  -F/opt/homebrew/Cellar/qt@5/5.15.8_3/lib -L/Users/laurent/sqlitestudio-osx-arm64/SQLiteStudio3/../output/SQLiteStudio -L/Users/laurent/sqlitestudio-osx-arm64/SQLiteStudio3/../../lib -L/usr/local/lib -lsqlite3 -framework QtQml -framework QtNetwork -framework QtCore -framework DiskArbitration -framework IOKit    
ld: warning: directory not found for option '-L/Users/laurent/sqlitestudio-osx-arm64/SQLiteStudio3/../../lib'
Undefined symbols for architecture arm64:
  "_sqlite3_enable_load_extension", referenced from:
      AbstractDb3<Sqlite3>::openInternal() in dbsqlite3.o
  "_sqlite3_load_extension", referenced from:
      AbstractDb3<Sqlite3>::loadExtension(QString const&, QString const&) in dbsqlite3.o
ld: symbol(s) not found for architecture arm64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[1]: *** [../../SQLiteStudio/libcoreSQLiteStudio.1.0.0.dylib] Error 1

```
